### ISSUE TITLE
Docker build error

### ISSUE THREAD

**Issue by teraaret** [postgis/docker-postgis#155]
> E: Version '2.5.2+dfsg-1~exp1.pgdg90+1' for 'postgresql-11-postgis-2.5' was not found
> E: Version '2.5.2+dfsg-1~exp1.pgdg90+1' for 'postgresql-11-postgis-2.5-scripts' was not found
> E: Version '2.5.2+dfsg-1~exp1.pgdg90+1' for 'postgis' was not found

I do not understand what means a '+dfsg-1~exp1.pgdg90+1' addition. What can I do wrong? Or maybe this version is not supported now?

**Comment by ImreSamu** [comment]
you have to run `./update.sh` for finding latest debian package version.

```
git clone https://github.com/appropriate/docker-postgis.git
cd docker-postgis/
./update.sh
git status
```


### diff 11-2.5/Dockerfile
```
$ git diff 11-2.5/Dockerfile
diff --git a/11-2.5/Dockerfile b/11-2.5/Dockerfile
index 381480c..b96a9cb 100644
--- a/11-2.5/Dockerfile
+++ b/11-2.5/Dockerfile
@@ -2,7 +2,7 @@ FROM postgres:11
 MAINTAINER Mike Dillon <mike@appropriate.io>
 
 ENV POSTGIS_MAJOR 2.5
-ENV POSTGIS_VERSION 2.5.2+dfsg-1~exp1.pgdg90+1
+ENV POSTGIS_VERSION 2.5.3+dfsg-2.pgdg90+1
 
 RUN apt-get update \
       && apt-cache showpkg postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR \
```

**Comment by ImreSamu** [comment]
> I do not understand what means a '+dfsg-1~exp1.pgdg90+1' addition.

like a debian package version numbers 
* https://unix.stackexchange.com/questions/96599/understanding-debian-dpkg-package-version-numbers 


> ...  for 'postgis' was not found

Some debian postgis packaging changed  ( probably related the new 3.0 version )
just remove the postgis line `postgis=$POSTGIS_VERSION \`   ; in theory - it should work.


```
...
       && apt-get install -y --no-install-recommends \
            postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR=$POSTGIS_VERSION \
            postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR-scripts=$POSTGIS_VERSION \
-           postgis=$POSTGIS_VERSION \      < ----  Delete this line
       && rm -rf /var/lib/apt/lists/*
...
```

**Comment by teraaret** [comment]
> you have to run `./update.sh` for finding latest debian package version.

Maybe I do not understand something obivious, but how can I run this script without postgres? (which not installed because of first error I wrote...)

**Comment by teraaret** [comment]
> just remove the postgis line `postgis=$POSTGIS_VERSION \` ; in theory - it should work.
Unfortunately, it doesn't

**Comment by teraaret** [comment]
I solved my problem with changing POSTGIS_VERSION to **2.5.3+dfsg-2.pgdg90+1**

Do not sure it will works as expected but in my case it works, and containers was build successfully.

**Comment by ImreSamu** [comment]
>>you have to run ./update.sh for finding latest debian package version.
> Maybe I do not understand something obivious, but how can I run this script without postgres?

simple bash  - it is working on ubuntu
* https://github.com/appropriate/docker-postgis/blob/master/update.sh

**Comment by ImreSamu** [comment]
if it is not working .. you have to check the https://github.com/appropriate/docker-postgis/pulls

**Comment by avilaton** [comment]
We ran into the same issue for 10-2.5 and even after running update.sh, the dockerfile fails to build
```
diff --git a/10-2.5/Dockerfile b/10-2.5/Dockerfile
index c2b78f5..ef413a7 100644
--- a/10-2.5/Dockerfile
+++ b/10-2.5/Dockerfile
@@ -2,7 +2,7 @@ FROM postgres:10
 MAINTAINER Mike Dillon <mike@appropriate.io>
 
 ENV POSTGIS_MAJOR 2.5
-ENV POSTGIS_VERSION 2.5.2+dfsg-1~exp1.pgdg90+1
+ENV POSTGIS_VERSION 2.5.3+dfsg-2.pgdg90+1
```
is the diff we see from update.sh.
Building results in a failure to locate the postgis package
`E: Version '2.5.3+dfsg-2.pgdg90+1' for 'postgis' was not found`
By looking at
http://apt.postgresql.org/pub/repos/apt/dists/stretch-pgdg/main/binary-amd64/Packages
One see that the following packages are available for this version-
```
Package: postgresql-10-postgis-2.5
Source: postgis-2.5
Version: 2.5.3+dfsg-2.pgdg90+1
...
Package: postgresql-10-postgis-2.5-scripts
Source: postgis-2.5
Version: 2.5.3+dfsg-2.pgdg90+1
...
Package: postgis
Version: 3.0.0+dfsg-2~exp1.pgdg90+1
```
so not really clear how this got screwed up.

**Comment by Desa007** [comment]
I have some different issue with build from Dpckerfile:
ERROR: unsatisfiable constraints:
  so:libgeos_c.so.1 (missing):
    required by: gdal-2.4.2-r1[so:libgeos_c.so.1]
  geos-dev (missing):
    required by: .build-deps-edge-0[geos-dev]
  proj-dev (missing):
    required by: .build-deps-edge-0[proj-dev]

**Comment by GammaGames** [comment]
@Desa007 see #157

**Comment by Desa007** [comment]
> @Desa007 see #157

Yes it's work. But now it fails with configure error:

configure: error: could not find libgeos_c - you may need to specify the directory of a geos-config file using --with-geosconfig

**Comment by thkwon1** [comment]
I also solved above problem 😄  Thank you


To solve problem, In your Docker file..

1. Change `POSTGIS_VERSION 2.5.2+dfsg-1~exp1.pgdg90+1` -> `POSTGIS_VERSION 2.5.3+dfsg-2.pgdg90+1`

2.  Delete `postgis=$POSTGIS_VERSION`

**Comment by frdrk345** [comment]
I tried to build 9.6-2.5
After running ./update.se, i got the same error as avilaton
( https://github.com/appropriate/docker-postgis/issues/155#issuecomment-557647557 )

I got docker to build the image by pointing to the revision of postgis that is available, 3.0.0+dfsg-2~exp1.pgdg90+1. Is this correct, or should there be a revision 2.5 of the postgis package?

    $ git diff HEAD~1
    diff --git a/9.6-2.5/Dockerfile b/9.6-2.5/Dockerfile
    index 652c851..e734433 100644
    --- a/9.6-2.5/Dockerfile
    +++ b/9.6-2.5/Dockerfile
    @@ -3,13 +3,14 @@ MAINTAINER Mike Dillon <mike@appropriate.io>
     
     ENV POSTGIS_MAJOR 2.5
     ENV POSTGIS_VERSION 2.5.3+dfsg-2.pgdg90+1
    +ENV PACKAGE_POSTGIS_VERSION 3.0.0+dfsg-2~exp1.pgdg90+1
     
     RUN apt-get update \
           && apt-cache showpkg postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR \
           && apt-get install -y --no-install-recommends \
                postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR=$POSTGIS_VERSION \
                postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR-scripts=$POSTGIS_VERSION \
    -           postgis=$POSTGIS_VERSION \
    +           postgis=$PACKAGE_POSTGIS_VERSION \
           && rm -rf /var/lib/apt/lists/*
     
     RUN mkdir -p /docker-entrypoint-initdb.d

**Comment by phillipross** [comment]
Hi all... we're currently working to get this repo updated to build current and supported images.  There will be changes upcoming to the update.sh script to make this all functional with the latest versions of the packages in debian repos, etc.

It should be possible to clone the master branch, run `make update build` which will first invoke update.sh to process the templates and resolve the ENV vars to the proper values, and second invoke docker to build the Dockerfiles.  Currently PR #166 is in-progress to fixup and address problems with this process, but the master branch should be working now for PostGIS 2.5.

I'm closing this issue for now and welcome you to re-test and file new issues against the master branch.  And feel free to follow the work on the PRs if you'd like.
